rfd/0097-teleport-connect-usage-metrics.md

```diff
@@ -0,0 +1,221 @@
+---
+authors: Grzegorz Zdunek (grzegorz.zdunek@goteleport.com)
+state: draft
+ ---
+
+# RFD 97 - Teleport Connect usage metrics
+
+## Required Approvers
+
+* Engineering:? @ravicious @zmb3
+* Product: @klizhentas @xinding33
+
+## What
+
+Collect downloads and usage metrics of Teleport Connect.
+
+## Why
+
+Currently, the team has no information on how many users download, install and use Teleport Connect on a daily basis.
+In order to effectively plan the development of the product, the team should also know what is the adoption of new
+features, which ones are the most popular and which are problematic for users.
+
+## Details
+
+Connect will have a stateless metrics service that will send events to an unauthorized gRPC endpoint, exposed by TSH
+daemon. The anonymization should be done in `tshd` (to be open on how it is done and what is sent further). Events will
+be sent to `prehog` directly.
+It was considered to use authorized endpoint provided by cluster's `Auth Service`, but it seems to not work well for
+Connect for a few reasons:
+
+- Some events do not belong to any cluster, like _app start_ event.
+- A user can use Connect while not logged in. For example, new tabs can be opened after the session expires.
+
+Since Connect will use endpoint that is not aware of the cluster context, it must provide some value that will
+differentiate events coming from multiple application instances. To do this, each event will have `distinct_id` field.
+It will be supplied with UUID generated by Connect, which will be generated once and stored in a file in the _App Data_
+directory, so it won't change between restarts.
+
+### User agreement
+
+On the start, Teleport Connect will ask user to opt in to volunteer anonymized metrics and usage-data with standard
+message "Are you OK sending anonymized usage data about Teleport Connect? This will help us to improve product".
+
+If the user refuses, no data will be collected.
+
+### How will collecting metrics support product development?
+
+In the initial version, it should help with getting answers to the following questions:
+
+#### How many unique users download and use Teleport Connect today?
+
+To answer the first part of the question, download counts from `goteleport.com/download` are needed.
+This task should be done by the marketing team.
+
+To calculate how many users use Teleport Connect on a daily basis, a metric like DAU (Daily Active Users) can be used.
```

Comment:
I think we should also track (at the auth server) how many users logged in with Teleport Connect. You can download, and even open Teleport Connect, but you aren't really _using_ it until you connect to a cluster.

This requires some sort of user agent so that auth knows it was connect that logged in and not `tsh` or the web UI. Can you include this in scope for the RFD? (how will you propagate a user agent to the backend)

Reply:
I think there is an easier way, Teleport connect can just emit anonymized metric after successful login.

Reply:
Collect an event after user successfully logs in.